libvast/test/detail/legacy_deserialize.cpp

```diff
@@ -0,0 +1,346 @@
+      },
+    },
+    {
+      "y",
+      legacy_record_type{
+        {"b", legacy_bool_type{}},
+      },
+    },
+  };
+
+  std::vector<char> buf;
+  CHECK_EQUAL(detail::serialize(buf, r), caf::none);
+  auto r2 = legacy_record_type{};
+  REQUIRE(ldes(buf, r2));
+  CHECK_EQUAL(r, r2);
+}
+
+TEST(deserialize_qualified_record_field) {
+  const auto field = qualified_record_field{
+    "zeek.conn",
+    record_field{
+      "conn.id",
+      legacy_address_type{},
+    },
+  };
+
+  std::vector<char> buf;
+  CHECK_EQUAL(detail::serialize(buf, field), caf::none);
+  auto field2 = qualified_record_field{};
+  REQUIRE(ldes(buf, field2));
+  CHECK_EQUAL(field, field2);
+}
+
+TEST(deserialize_ids) {
+  auto i = ids{};
+  i.append_bits(true, 20);
+  i.append_bits(false, 5);
+  i.append_bits(true, 1);
+  std::vector<char> buf;
+  CHECK_EQUAL(detail::serialize(buf, i), caf::none);
+  auto i2 = ids{};
+  REQUIRE(ldes(buf, i2));
+  CHECK_EQUAL(i, i2);
+}
+
+auto to_addr_view(std::string_view str) {
+  return make_data_view(unbox(to<address>(str)));
+}
+
+TEST(deserialize_time_synopsis) {
+  using vast::time;
+  const time epoch;
+  factory<synopsis>::initialize();
+  auto ts = factory<synopsis>::make(legacy_time_type{}, caf::settings{});
+  REQUIRE_NOT_EQUAL(ts, nullptr);
+  ts->add(time{epoch + 4s});
+  ts->add(time{epoch + 7s});
+  std::vector<char> buf;
+  CHECK_EQUAL(detail::serialize(buf, ts), caf::none);
+  auto ts2 = factory<synopsis>::make(legacy_time_type{}, caf::settings{});
+  REQUIRE(ldes(buf, ts2));
+  CHECK_EQUAL(*ts, *ts2);
+}
+
+TEST(deserialize_bool_synopsis) {
+  factory<synopsis>::initialize();
+  auto bs = factory<synopsis>::make(legacy_bool_type{}, caf::settings{});
+  REQUIRE_NOT_EQUAL(bs, nullptr);
+  std::vector<char> buf;
+  CHECK_EQUAL(detail::serialize(buf, bs), caf::none);
+  auto bs2 = factory<synopsis>::make(legacy_bool_type{}, caf::settings{});
+  REQUIRE(ldes(buf, bs2));
+  CHECK_EQUAL(*bs, *bs2);
+}
+
+TEST(deserialize_address_synopsis) {
+  factory<synopsis>::initialize();
+  auto lat
+    = legacy_address_type{}.attributes({{"synopsis", "bloomfilter(1,0.1)"}});
+  auto as = factory<synopsis>::make(lat, caf::settings{});
+  REQUIRE_NOT_EQUAL(as, nullptr);
+  as->add(to_addr_view("192.168.0.1"));
+  std::vector<char> buf;
+  CHECK_EQUAL(detail::serialize(buf, as), caf::none);
+  auto as2 = factory<synopsis>::make(lat, caf::settings{});
+  REQUIRE(ldes(buf, as2));
+  CHECK_EQUAL(*as2, *as);
+}
+
+TEST(deserialize_string_synopsis) {
+  factory<synopsis>::initialize();
+  auto lst
+    = legacy_string_type{}.attributes({{"synopsis", "bloomfilter(1,0.1)"}});
+  auto ss = factory<synopsis>::make(lst, caf::settings{});
+  REQUIRE_NOT_EQUAL(ss, nullptr);
+  ss->add(std::string_view{"192.168.0.1"});
+  std::vector<char> buf;
+  CHECK_EQUAL(detail::serialize(buf, ss), caf::none);
+  auto ss2 = factory<synopsis>::make(lst, caf::settings{});
+  REQUIRE(ldes(buf, ss2));
+  CHECK_EQUAL(*ss, *ss2);
+}
```

Comment:
Not a request to change anything, just wanted to say thanks for writing this really extensive test suite. It's great to see! 🚀

Reply:
Writing test cases was the least I could do.